src/utils/object.js

```diff
@@ -1,15 +1,29 @@
+import { flatten } from './array';
+
 export const isObject = obj => Object.prototype.toString.call(obj) === '[object Object]';
 
-export const compactObject = (obj) => {
+const iterateeWithCondition = (
+  obj,
+  condition = () => true,
+  iteratee = iterateeObj => iterateeObj
+) => {
   if (!isObject(obj)) {
     throw new TypeError(`${obj} is not an Object`);
   }
 
   return Object.keys(obj).reduce((memo, key) => {
-    if (obj[key]) {
-      return Object.assign(memo, { [key]: obj[key] });
+    if (condition(obj, key)) {
+      return Object.assign(memo, { [key]: iteratee(obj[key], key) });
     }
 
     return memo;
   }, {});
 };
+
+export const compactObject = obj => iterateeWithCondition(obj, (item, key) => !!item[key]);
+
+export const mapObject = (obj, iteratee) => iterateeWithCondition(obj, () => true, iteratee);
+
+export const pick =
+  (obj, ...keys) =>
+    iterateeWithCondition(obj, (item, key) => flatten(keys).indexOf(key) > -1);
```

Comment:
no need to run flatten(keys) every time

Reply:
flatten(keys).reduce((memo, key) => {
  if (obj.hasOwnProperty(key)) {
    return Object.assign(memo, {[key]: obj[key]});
  }
  return memo;
}, {})

Reply:
```javascript
export const pick = (obj, ...keys) => {
  const flattendKeys = flatten(keys);

  return iterateeWithCondition(obj, (item, key) =>
    (searchKeys => searchKeys.indexOf(key) > -1)(flattendKeys));
};
```